main.go

```diff
@@ -108,6 +110,16 @@ func main() {
 		tlsClientCertificates = append(tlsClientCertificates, cert)
 	}
 
+	var tlsCaCertificates *x509.CertPool
+	if *tlsCaCertFile != "" {
+		caCert, err := ioutil.ReadFile(*tlsCaCertFile)
+		if err != nil {
+			log.Fatalf("Couldn't load TLS Ca certificate, err: %s", err)
+		}
+		tlsCaCertificates := x509.NewCertPool()
```

Comment:
The variable `tlsCaCertificates` is re-declared here and shadows the variable from line 113 which means you're never successfully picking up a CA cert.
Change `:=` to `=` and you should be fine.

Reply:
Fixed

Reply:
Thanks, this looks good now.